service/notification/NotificationProcessor.php

```diff
@@ -335,4 +350,40 @@ private function isCurrentOrderStatusANonFinalStatus($currentOrderStatus)
 
         return false;
     }
+
+    /**
+     * @param $notification
+     * @param OrderCore $order
+     * @return bool
+     * @throws \Exception
+     */
+    private function validateWithCartAndOrder($notification, OrderCore $order)
+    {
+        $cart = \Cart::getCartByOrderId($order->id);
+
+        if (!\Validate::isLoadedObject($cart)) {
+            return false;
```

Comment:
pls add a log here so we know what went wrong and not the actual validation failed